h/api/models/test/document_test.py

```diff
@@ -119,6 +122,416 @@ def test_document_find_or_create_by_uris_no_results():
+
+    def test_it_sets_value(self, DocumentMeta):
+        """If there's an existing DocumentMeta it should update its value."""
+        existing_document_meta = self.mock_document_meta()
+        existing_document_meta.value = 'old value'
+        DocumentMeta.query.filter.return_value.one_or_none\
+            .return_value = existing_document_meta
+
+        document.create_or_update_document_meta(
+            db=mock_db(),
+            claimant='http://example.com/claimant',
+            claimant_normalized='http://example.com/claimant_normalized',
+            type='title',
+            value='new value',
+            document=mock.Mock(),
+            created=yesterday(),
+            updated=now(),
+        )
+
+        assert existing_document_meta.value == 'new value'
+
+    def test_sets_updated(self, DocumentMeta):
+        """If there's an existing DocumentMeta it should update its updated."""
+        existing_document_meta = self.mock_document_meta()
+        existing_document_meta.updated = yesterday()
+        DocumentMeta.query.filter.return_value.one_or_none\
+            .return_value = existing_document_meta
+        now_ = now()
+
+        document.create_or_update_document_meta(
+            db=mock_db(),
+            claimant='http://example.com/claimant',
+            claimant_normalized='http://example.com/claimant_normalized',
+            type='title',
+            value='new value',
+            document=mock.Mock(),
+            created=yesterday(),
+            updated=now_,
+        )
+
+        assert existing_document_meta.updated == now_
+
+    def test_it_logs_warning(self, DocumentMeta, log):
+        """
+        It should warn on document mismatches.
+
+        It should warn if there's an existing DocumentMeta with a different
+        Document.
+
+        """
+        document_one = mock_document()
+        document_two = mock_document()
+        existing_document_meta = self.mock_document_meta(document=document_one)
+        DocumentMeta.query.filter.return_value.one_or_none\
+            .return_value = existing_document_meta
+
+        document.create_or_update_document_meta(
+            db=mock_db(),
+            claimant='http://example.com/claimant',
+            claimant_normalized='http://example.com/claimant_normalized',
+            type='title',
+            value='new value',
+            document=document_two,
+            created=yesterday(),
+            updated=now(),
+        )
+
+        assert log.warn.call_count == 1
+
+    def mock_document_meta(self, document=None):
+        class DocumentMeta(object):
+            def __init__(self):
+                self.claimant_normalized = None
+                self.type = None
+                self.value = None
+                self.created = None
+                self.updated = None
+                self.document = document
+                self.id = None
+                self.document_id = None
+        return mock.Mock(spec=DocumentMeta())
```

Comment:
Added a comment to clarify this. We can't use the real `DocumentMeta` as a spec because in the test that class is patched with a mock.